### DISCUSSION TITLE
Best Method for creating a Strong, Stable, Selfhealing Zigbee Mesh/External Omnidirectional Antennas

### DISCUSSION THREAD

**Discussion by LordNex** [tube0013/tube_gateways#163] [Ideas]
I was just wondering if anyone has had any success with adding a different antenna to their coordinator. I've seen a few directional ones that will send a Zigbee signal pretty far away with line of sight. but im more interested in putting something in the rafters of my home that will better permeate all the areas. I don't have a l;arge home. 3 bedroom 1.5 bath on 2 acers of land. Inside I have 2 Aruba IAP Access Points in a cluster with an additional outdoor unit and a wifi 6 AP. everything is VLAN'd out and controlled by Managed Switches and a Firewalla Gold router IPS\IDS. 

Prior to using Tube's ERF32 PoE I was just using a Nortek HUBZ1 on an extinction cable. but now that I have access to an SNA port im wondering if there wouldn't be a better antenna that the little plastic one. Anyone have  ideas or messed with any of this? ANy links to amazon products would be great. 

One last thing, i have it put way upnor id look myself, Is that an SNA with the central pin or without? Makes a difference when combining it to the proper antenna.

**Comment by LordNex** [comment]
I added an old indoor cell repeater antenna and it seemed to do much better than the dinky thing that comes with it. I think I'm just going to get a good self powered antenna if I can find one that will boost my Zigbee for more connections, eventually reaching all the areas I need on my 2 acres. If anyone knows of anything like that, I would greatly appreciate it

**Comment by MattWestb** [comment]
You is trying building one star network like your WiFi is working and not one mesh network that can rerouting if having local RF problems.
The right way to going is adding good router devices for extending your network and not one some _kilowatt_ output power of your coordinator that you is only getting problems in the long run with radio interference you cant going around with one star network but one good working mesh network can do.
Also your RF links is being unsymmetrical (one sleeping end device is using 1/10 of the power of one router) and in the end the sleepers can _hearing_ the coordinator but the coordinator cant _hearing_ the end device = not working well at all.

**Reply by LordNex** [reply]
I think I understand what your meaning and I'm trying to do just that. But u alsobgavr a huge network of over 70 devices, about half of which are wireless and I'd say 75% of the wireless are IoT devices that can only run on 2.4ghz Wi-Fi. In tan some tests with different external antennas and ended up purchasing an omnidirectional pure copper antenna and intend to install in it my attic space. My home only had one level but is a long rectangle ranch style home on 2 acres of land. Ultimately I need those sensors to be anywhere in that 2 acres but something's ill probably have to ESPHome build of want any ability to connnect. 

I just put in an order for 4 E12 Singled bulbs for my dining room and 4 standard socket singlade bulbs for my back bedroom, and 2 large outdoor floodlamps for in front of my garage. So far I have a mains powered device acting as a router ( they show up using the LEDVANCE quirk and show up in ZHA as a router. That'll bring the grand total of mains powered bulbs that are acting as repeaters up to 20. I also have Aqara door and window sensors, water leak sensors, a yet to be installed main weather valve shut-off (we live off a well). I'm not using anything fancy. I do have the ZHA advanced toolkit from HACS installed but have no idea how to use it. 

What's the best method to create a strong Mesh Network? Should I remove everything and start by connecting each bulb directly to the coordinator? Or should I tie one to the coordinator and then next build to it and so on. As far as the visualization of the network in ZHA, the topology is constantly shifting. So I know some of the mesh is working. I've avoided using other tools such as ZHAtoMQTT as I don't see a benefit. If you know of a reason why I should change that thought I'd be happy to hear what you know.

I would think this should be pretty easy as I don't live anywhere near other people or anything. I'm in a completely rural area that is a cell phone black hole. I was able to get some cell coverage by installing a very expensive outdoor to indoor repeater/amplifier that is hanging off a 70 foot tower attached to my house that I used to get my Internet through. Now I'm on Starlink and it's a pain to constantly try and deal with their CGNAT. I have a fully deployed IPv6 network along side my IPv4 in the /56 subnet. But until more networks start adopting IPv6 I end up having to route through proxies or VPN tunnels. Luckily I have a very robust network which is my main bread and butter. Im primarily a Microsoft Network guy who also knows a lot about networking in general. But I've only had about 5 years working with Zigbee and ZWave. Haven't touched matter "yet". 

My main issue is I keep getting end devices that are battery powered drop out of the mesh every now and then. No real pattern that I can tell but it is usually the same devices. Some of that I'm sure I can attribute to cheap manufacturing or something similar but sometimes I won't have any problems for months. Then like 3 end devices will randomly just go unavailable and I have to repair them to get it back online. 

So yea please explain or point me in a direction of what's the best way to build a solid Zigbee mesh because Google and YouTube are back and forth on what they think is the best way. Or it's advice from a country that doesn't use the same brand of hardware so I can't tell what's causing it.

**Reply by LordNex** [reply]
![image](https://github.com/tube0013/tube_gateways/assets/9270948/840e8c15-c365-4f7a-9d91-545fac9c7b89)

**Reply by LordNex** [reply]
Again. If someone can explain the best method of connecting everything together so that the mesh is solid and self healing. Please let me know. Originally I figured I'd just pair everything with the coordinator but that seems like it's worse. So should I connect all the mains powered devices to the coordinator first, then add the end points to the router/bulbs? 

If I can't get it to work this way Aqara has a Zigbee network booster for about 40 bucks. But I'm going for wait for the 12 or so new SYLVANIA lights to come in and see if that helps some of the dead zones

**Comment by tube0013** [comment]
connect mains powered routing devices first. (Sengled bulbs are not routers). start with the routing devices closest to coordinator and move out ward from there. doesn't really matter how you add the routing devices, as they will pick their routes and connect to each other and coordinator if possible. when you start adding end devices, pick the router closest and add through that. They may not stick if they are zigbee 3.0 and find a better path that may be physically further but clearer for RF, but it's a good way to start if starting from scratch.

**Reply by LordNex** [reply]
These are sylvanya bulbs. I'm using a combo of these 3 


My primary bulb in my house and show up as a router in ZHA

![image](https://github.com/tube0013/tube_gateways/assets/9270948/ea824e79-edd2-44ee-a304-179ca606fc2d)

[SYLVANIA SMART Zigbee Full Color A19 LED Bulb for SmartThings Echo Plus, 10.5W, Hub Needed for Amazon Alexa and Google Assistant - 4 Count(Pack of 1)](SYLVANIA SMART Zigbee Full Color A19 LED Bulb for SmartThings Echo Plus, 10.5W, Hub Needed for Amazon Alexa and Google Assistant - 4 Count(Pack of 1) https://a.co/d/j0xOOzn)

I needed an [E12 bulb and grabbed these](Limited-time deal: Sengled Zigbee Smart Light Bulbs, Smart Hub Required, Work with SmartThings and Echo with built-in Hub, Voice Control with Alexa and Google Home, Color Changing E12 Candelabra Light Bulbs 40W 2 Pack https://a.co/d/cNlb9KW). These are the only sengled bulbs that'll will be going in. I have no idea how these will act and luckily with Amazon, if I don't like them I can send them back for free and get something else. 

And finally I thought I'd push the envelope, may have to put an LEDVANCE nearest my garage. But i picked up a set of these [BR30 flood lights](Sengled Zigbee BR30 Smart Bulb, Works with SmartThings and Echo with Built-in Hub, Voice Control with Alexa and Google Home, Hub Required, Dimmable Flood Light Bulb, Soft White 2700K, 2 Pack https://a.co/d/jhf6RTW) for the front of my garage. But on sale $10 for a 2 pack I didn't think I could go too wrong. 

It has a simple PIR sensor that if it sees movement, it kicks them on. But at the bulb level not at the power switch so that sensor may need to be disabled and an ESP8266 and PIR or dual 533mhz RF PIRs to tied to rule out false positives as I still use some of my Sonoff RF bridges flashed with tazmota for those more far off sensors. They're just a bit more complicated but give me an excuse to break out the solder gun. 

But again these are sengled bulbs so again I'll have to test and report back. My garage is about 75 foot in front of my house so it may require a separate coordinator or a Zigbee Powered Repeater and directional antenna to pull off. 

Thanks for the advice. I might just delete the ZHA db and integration all together. Reflash the coordinator with the latest bin from your repo, and start connecting the LEDVANCE bulbs first and see if these newer sengled are any better. I also picked up one of these. I'd prefer to keep the PoE board out of the harsh heat of my attic in the summer so I picked up this for pretty cheap and plan on using it as the external antenna. I could boost the dB but would probably just increase the noise floor of the existing 2.4ghz. 

[High Gain 8dBi SMA Wireless Antenna for 2.4GHz WiFi Router Network Card ZigBee (10 Meters Long) https://a.co/d/abptbzv]()

I haven't read much on matter yet but I'm starting to see devices like those on Amazon now too. I need a new protocol to learn like I need another hole in my head. 

Thanks for taking the time to help me get this figured out. Hopefully this will work with some of the extra bulbs

**Comment by LordNex** [comment]
> connect mains powered routing devices first. (Sengled bulbs are not routers). start with the routing devices closest to coordinator and move out ward from there. doesn't really matter how you add the routing devices, as they will pick their routes and connect to each other and coordinator if possible. when you start adding end devices, pick the router closest and add through that. They may not stick if they are zigbee 3.0 and find a better path that may be physically further but clearer for RF, but it's a good way to start if starting from scratch.

Yep you were dead on right. These sengled bulbs (I got my E12s today) come up as end points. So I'm not sure if I should send them back and see if I can find E12s built by LEDVANCE or something. Is there a specific brand you'd suggest?  

My Sylvanya bulbs are LEDVANCE and show up as routers per the screenshot. I think they can only handle 6 connections at a time regardless of that's an end point or router. I also picked up a few more aquara door/window sensors. So far it's been kind of hit or miss with those but I think a lot of it is the mesh not getting a good RF RSSI from them

**Reply by LordNex** [reply]
@tube0013 Ok well I guess I'm going to pack up these Sengled E12 base bulbs. They seem to be holding ok as end points though. But I obviously need more hubs. I cannot believe how hard it is to find bulbs that can handle more than one connection or have 3.0 support. 

So I'm thinking of replacing them with these. They are leaps and bounds more expensive but if it works. Then it's worth the money. 

LEDEPLY Zigbee A15 Smart Bulbs, Compatible with Hue*, Alexa, Google & ConBee (Hub Required),E12 Color Changing, Tunable White 2700K-6500K, Globe Shape, G45 Ceiling Fan Light Bulbs 2Pack https://a.co/d/cevTV0K

I base everything off the W.A.F. Or Wife Approval Factor. And she just wants it to work the first time every time which is turning out to be a nightmare when it comes to Zigbee. I'm also most wondering if I shouldn't ditch Zigbee all together and just run everything off WiFi. I run 2 Aruba Campus Grade IAP335s and 1 x Outdoor IAP275. These are made for huge networked campuses and run off a self hosting controller that they create out of their own mesh. They are like $1500 new but you can usually pick them up cheaper online eBay used. I recently installed 75 in a school where we had one in each classroom and one every 25 foot in the halls. Each is rated to handle 75+ devices with dual 5ghz radios each with a 6x6 MU-MIMO and 12 antennas. 

If you have any other suggestions on routable Zigbee bulbs other than the standard base. I'm looking for E12, BR30, and GR24 "Tesla" bulbs. I sure would appreciate it. Not finding much that I feel I can trust on Amazon so I'm interested in what you think about the above bulb.

**Reply by LordNex** [reply]
@tube0013 Could you suggest some bulbs accessible in the USA, preferably on Amazon that would be a good repeater for the various common sizes? I think it would help us all to know what to look for in building a good mesh network. All I could find was above. 

Could you please provide some brands that would be best?

Common US Bulb Sizes:
A19
E12
BR30
GU24

Thanks a bunch for your help!
